agent/consul/gateways/controller_gateways.go

```diff
@@ -147,10 +195,11 @@ func (r *apiGatewayReconciler) cleanupBoundGateway(_ context.Context, req contro
 	for _, modifiedRoute := range removeGateway(resource, routes...) {
 		routeLogger := routeLogger(logger, modifiedRoute)
 		routeLogger.Trace("persisting route status")
-		if err := r.updater.Update(modifiedRoute); err != nil {
+		if err := r.updater.UpdateWithStatus(modifiedRoute); err != nil {
```

Comment:
There was some pointer shenanigans happening here where the code was modifying object returned by the storage layer, and as soon as I added deepcopies, the logic was no longer behaving as expected. Updating with the status flag fixed this.